sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/StringBasedCosmosQuery.java

```diff
@@ -55,24 +55,29 @@ public Object execute(final Object[] parameters) {
         String expandedQuery = query;
         List<SqlParameter> sqlParameters = new ArrayList<>();
         for (int paramIndex = 0; paramIndex < parameters.length; paramIndex++) {
-            Parameter queryParam = getQueryMethod().getParameters().getParameter(paramIndex);
-            if (parameters[paramIndex] instanceof Collection) {
-                List<String> expandParam = ((Collection<?>) parameters[paramIndex]).stream()
-                    .map(Object::toString).collect(Collectors.toList());
-                List<String> expandedParamKeys = new ArrayList<>();
-                for (int arrayIndex = 0; arrayIndex < expandParam.size(); arrayIndex++) {
-                    String paramName = "@" + queryParam.getName().orElse("") + arrayIndex;
-                    expandedParamKeys.add(paramName);
-                    sqlParameters.add(new SqlParameter(paramName, toCosmosDbValue(expandParam.get(arrayIndex))));
-                }
-                expandedQuery = expandedQuery.replaceAll("@" + queryParam.getName().orElse(""), String.join(",", expandedParamKeys));
-            } else {
-                if (!Pageable.class.isAssignableFrom(queryParam.getType())
-                    && !Sort.class.isAssignableFrom(queryParam.getType())) {
-                    sqlParameters.add(new SqlParameter("@" + queryParam.getName().orElse(""), toCosmosDbValue(parameters[paramIndex])));
+                Parameter queryParam = getQueryMethod().getParameters().getParameter(paramIndex);
+                if (!queryParam.getName().isEmpty()) {
+                    String inParamCheck = "ARRAY_CONTAINS(@" + queryParam.getName().get();
+                    if (parameters[paramIndex] instanceof Collection
+                        && expandedQuery.toString().toUpperCase().replaceAll("\\s", "")
```

Comment:
Would also like to see some comments here - I think I get what you are doing - but a sample with some comments explaining the flow would help to make it easier to still read and understand the code in a few weeks.

Reply:
Usually also string comparison is safer with lower case than with upper case - because there is no upper case letter for all characters - german special charatcer 'ß' for example has not corresponding upper case char

Reply:
Thanks for the feedback!